### DISCUSSION TITLE
How to upload an image from a folder | Provider BlackBox

### DISCUSSION THREAD

**Discussion by xxxpabloxxx** [xtekky/gpt4free#2256] [Q&A] [answered]
In */site-packages/g4f/Provider/BlackBox.py
![image](https://github.com/user-attachments/assets/7d157356-03a7-4b24-a5d2-37be109b3159)

**_in code python_** 
**_I use aiogram_**

**_Downloading an image and converting to base64 and sending a request_**
![image](https://github.com/user-attachments/assets/bffcec30-84e8-4397-ae1d-18a2ef369e45)

In what format should I transfer the image and am I doing it correctly? **BlackBoxAI** is able to process images

**Comment by Parthsadaria** [comment, marked as answer]
Put Image Inside Messages like this 
```         
messages=[
                   {
                    "role": "user",
                    "content": f"{prompt}",
                    "data": {
                        "fileText": "", # file name
                        "imageBase64": base64_image
                    }
                } 
            ]
```

**Reply by xxxpabloxxx** [reply]
Thanks man!
